From fb59986a15ca93346a1e2f4cbb0c90804dc0eb40 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Wed, 28 Aug 2024 18:51:21 +0200
Subject: [PATCH] Add explicit WITH_XFS option, be more verbose

Automagic dependencies must be avoided. When using other mechanisms than
find_package, let's add control through explicit CMake options.

This does not change default behaviour since WITH_XFS is default enabled,
and lack of xfsprogs continues to be non fatal.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b177d96a..34845fa7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,13 +90,19 @@ add_definitions(
 )
 add_definitions(-DTRANSLATION_DOMAIN="libktorrent6")
 
-#XFS prealloc
-check_include_files(xfs/xfs.h HAVE_XFS_XFS_H)
-if(HAVE_XFS_XFS_H)
-    set(CMAKE_EXTRA_INCLUDE_FILES xfs/xfs.h)
-    check_type_size(__u64 HAVE___U64)
-    check_type_size(__s64 HAVE___S64)
-    set(CMAKE_EXTRA_INCLUDE_FILES)
+option(WITH_XFS "Support for XFS specific disk preallocation" ON)
+if(WITH_XFS)
+    #XFS prealloc
+    check_include_files(xfs/xfs.h HAVE_XFS_XFS_H)
+    if(HAVE_XFS_XFS_H)
+        set(CMAKE_EXTRA_INCLUDE_FILES xfs/xfs.h)
+        check_type_size(__u64 HAVE___U64)
+        check_type_size(__s64 HAVE___S64)
+        set(CMAKE_EXTRA_INCLUDE_FILES)
+        message(STATUS "xfsprogs found, building with XFS specific disk preallocation support")
+    else()
+        message(STATUS "xfsprogs NOT found, XFS specific disk preallocation unavailable")
+    endif()
 endif()
 
 # check for 64 bit file I/O functions
-- 
GitLab

